src/settings.js

```diff
@@ -0,0 +1,249 @@
+ * requests to the tx-mining-service.
+ * @property {string|null} [atomicSwapService] - The atomic swap service url to
+ * use, it not present we will use the default.
+ * @property {Record<string, MultiSigConfig>} [multisig] - The multisig config object
+ * @property {string} [tokenUid] - The default token.
+ * @property {number|null} [gapLimit] - a custom gap limit, if not present we
+ * will use the default.
+ * @property {number|null} [connectionTimeout] - Websocket connection timeout in ms.
+ * @property {boolean} [allowPassphrase] - If we allow the use of passphrases
+ * when starting a wallet.
+ * @property {boolean} [confirmFirstAddress] - If true the user will have to
+ * provide the address at index 0 of the wallet he is trying to interact with.
+ * @property {string[]} [enabled_plugins] - The plugins we should start when
+ * the service starts up.
+ * @property {Record<string, PluginConfig>} [plugin_config] - The custom plugin configuration.
+ */
+
+/** @type {Configuration} */
+let _config = null;
+let started = false;
+const configPath = './config.js';
+
+/**
+ * Import the config and return the contents, removing from the `default` key.
+ * @returns {Promise<Configuration>}
+ */
+async function _importConfig() {
+  return (await import(configPath)).default;
+}
+
+/**
+ * Returns the config object.
+ * @returns {Configuration}
+ */
+function getConfig() {
+  if (_config) {
+    return _config;
+  }
+  // Configuration cannot be read at the moment.
+  throw new errors.UnavailableConfigError();
+}
+
+/**
+ *
+ */
+async function setupConfig() {
+  if (started) {
+    // This is a server error, so the default middleware will handle it correctly.
+    throw new Error('Cannot setup the configuration twice.');
+  }
+
+  _config = await _importConfig();
+  started = true;
+}
+
+async function _stopAllWallets() {
+  // stop all wallets
+  for (const wallet of initializedWallets.values()) {
+    await wallet.stop({ cleanStorage: true, cleanAddresses: true });
+    wallet.conn.removeAllListeners();
+  }
+
+  // Remove from initialized wallets
+  const keys = Array.from(initializedWallets.keys());
+  for (const walletId of keys) {
+    initializedWallets.delete(walletId);
+    await removeAllWalletProposals(walletId);
+  }
+}
+
+/**
+ * Analize the configuration changes and act accordingly
+ * @param {Configuration} oldConfig
+ * @param {Configuration} newConfig
+ */
+async function _analizeConfig(oldConfig, newConfig) {
+  // TODO: analize newConfig to check that we have a good config
+
+  // Checking changes in the fields:
+  // http_post, http_bind_address, http_api_key, consoleLevel, httpLogFormat, enabled_plugins
+  if (
+    oldConfig.http_port !== newConfig.http_port
+    || oldConfig.http_bind_address !== newConfig.http_bind_address
+    || oldConfig.http_api_key !== newConfig.http_api_key
+    || oldConfig.consoleLevel !== newConfig.consoleLevel
+    || oldConfig.httpLogFormat !== newConfig.httpLogFormat
```

Comment:
Because these are configured when creating the express app and we have not tested a way to change it while the app is running.
If we find a way to change this configuration while the app is running we can remove it from here and add a case to change this configuration in the future but since it cannot be changed I think that we should stop the app.
This way the user does not think his config change actually worked when it did not.